CodeListLibrary_project/clinicalcode/api/views/PhenotypeWorkingSet.py

```diff
@@ -27,6 +29,341 @@
 
 from drf_yasg.utils import swagger_auto_schema
 
+#------------------ phenotype working sets --------------------------------
+@swagger_auto_schema(method='get', auto_schema=None)
+@api_view(['GET'])
+def phenotypeworkingsets(request, pk=None):
+    '''
+        Get the API output for the list of my Phenotype Working Sets.
+    '''
+    search = request.query_params.get('search', None)
+
+    if pk is not None:
+        workingset_id = pk
+    else:
+        workingset_id = request.query_params.get('id', None)
+
+    tag_ids = request.query_params.get('tag_ids', '')
+    collection_ids = request.query_params.get('collection_ids', '')
+    owner = request.query_params.get('owner_username', '')
+    author = request.query_params.get('author', '')
+    show_only_my_workingsets = request.query_params.get('show_my_ph_workingsets', "0")
+    show_deleted_workingsets = request.query_params.get('show_deleted_ph_workingsets', "0")
+    ws_brand = request.query_params.get('brand', "")
+    do_not_show_versions = request.query_params.get('do_not_show_versions', "0")
+
+    data_sources = request.query_params.get('data_source_ids', '')
+    start_date_range = request.query_params.get('start_date', '')
+    end_date_range = request.query_params.get('end_date', '')
+    
+    start_date_query, end_date_query = False, False
+    try:
+        start_date_query = make_aware(datetime.datetime.strptime(start_date_range, '%Y-%m-%d'))
+        end_date_query = make_aware(datetime.datetime.strptime(end_date_range, '%Y-%m-%d'))
+    except ValueError:
+        start_date_query = False
+        end_date_query = False
+    
+    # ensure that user is only allowed to view/edit the relevant workingsets
+    workingsets = get_visible_phenotypeworkingsets(request.user)
```

Comment:
we don't use obj ORM here.
new working sets have a centralised search function in db_util which must be used in all related work 
get_visible_live_or_published_phenotype_workingset_versions()
https://github.com/SwanseaUniversityMedical/concept-library/blob/12e939dba81b5fe63853b8267e7120154ba97b67/CodeListLibrary_project/clinicalcode/db_utils.py#L4473

Reply:
please don't adapt anything from the old working set

Reply:
When we went over it last week we walked through the current API together in a Huddle which I which I aligned the original code with what we discussed, apologies - looks like I misunderstood what you were saying! I've since removed the previous code and we now rely on the ```get_visible_live_or_published_phenotype_workingset_versions``` method

Reply:
I need to implement the update/create endpoint and then the PR should be ready for review